lib/AST/ArchetypeBuilder.cpp

```diff
@@ -1030,6 +1036,37 @@ auto ArchetypeBuilder::resolveArchetype(Type type, PotentialArchetype *basePA)
   return nullptr;
 }
 
+auto ArchetypeBuilder::FIXMEresolveArchetypeOrTypeThroughTypeAlias(
+    Type type, PotentialArchetype *basePA)
+    -> llvm::PointerUnion<PotentialArchetype *, Type> {
+  auto pa = resolveArchetype(type, basePA);
+  if (!pa)
+    return type;
+
+  pa = pa->getRepresentative();
+  if (!pa->getParent() || !pa->getTypeAliasDecl())
```

Comment:
I think the only remaining direct callers are either `addSameTypeRequirement`, or are in places where equating to the type alias PA directly seems correct, specifically: potential archetype typo correction, equating any new PAs inside `getNestedType` itself, and recursively to merge the PAs.

However, having it inside `addSameTypeRequirementBetweenArchetypes` does seem like it might be nicer.